Source/Editor/Gizmo/TransformGizmo.cs

```diff
@@ -50,6 +50,44 @@ public TransformGizmo(IGizmoOwner owner)
         {
         }
 
+        /// <summary>
+        /// Helper function, recursivily finds the Prefab Root of node or null
+        /// </summary>
+        /// <param name="node">The node from which to start</param>
+        /// <returns></returns>
+        public ActorNode GetPrefabRootInParent(ActorNode node)
+        {
+            if (!node.HasPrefabLink)
+                return null;
+            if (node.Actor.IsPrefabRoot)
+                return node;
+            else if (node.ParentNode is ActorNode parAct)
+                return GetPrefabRootInParent(parAct);
+            else
+                return null;
+        }
+
+        /// <summary>
+        /// Recursively walks up from the node up to ceiling node(inclusive) or selection(exclusive)
+        /// </summary>
+        /// <param name="node">The node from which to start</param>
+        /// <param name="ceiling">The ceiling(inclusive)</param>
+        /// <returns></returns>
+        public ActorNode RecursiveWalkUpAndFindSomethingBeforeSelectedOrSupplied(ActorNode node, ActorNode ceiling)
```

Comment:
That's probably the longest method name I've ever seen 😂 

At the very least, we can drop the `Recursive`, since that's an implementation detail and replace the `Something` with `ActorNode`

Reply:
renamed